### DISCUSSION TITLE
The change to API 29 / Android 10, current consensus?

### DISCUSSION THREAD

**Discussion by patham9** [termux/termux-app#3382] [General]
An older wiki entry contained more information on this issue: https://github.com/termux/termux-packages/wiki/Termux-and-Android-10/5d899145ab70caa6e484609baf6c354651150230 compared to the current one.

There it was mentioned that Termux will migrate to using apk's for packages, also saying it will make it more inconvenient for users.
Sadly, to my understanding it will not just make it more inconvenient, but it will ultimately break Termux for users like me who use Termux to compile and execute C code on my Android phone without having to go through the clunkly and time-consuming official development tools on another machine such as a PC.

Now I'm also no fan of WASM of course, having seen that performance is still 3x as slow for most programs, not even close to the numbers claimed in terms of how close it "should be" to native speed. But what about the PRoot solution, did this not seem very promising? 

What is the current consensus of how Termux will proceed in the near and long-term future in this regard?
How will we simply invoke gcc and execute the generated binary file in the future through Termux?

**Comment by agnostic-apollo** [comment]
Duplicate of #2155
